### ISSUE TITLE
biber tool crashes with 'Can't call method "set_field" on an undefined value'

### ISSUE THREAD

**Issue by tnull** [plk/biber#374]
When I run `biber --tool --tool-noremove-missing-dependants`, it crashes and does not process the bib file entirely: 

```
$ biber --tool --tool-noremove-missing-dependants --configfile sort.conf --outfile out.bib in.bib
...
Can't call method "set_field" on an undefined value at /var/folders/p1/rrlkj5416hxfvqfz_f44n7jw0000gn/T/par-65726f/cache-bc6986241d41958a757fc1df8a704e6d22b2f953/inc/lib/Biber.pm line 1557.
```

Exactly this configuration used to work for many versions, however, since some time it stopped. Maybe the issue #271 has been somehow re-introduced? 

```
$ biber --version
biber version: 2.16
```

**Comment by plk** [comment]
Can you give a sample `.bib` file that gives this error? Does it occur without the `--configfile`?

**Comment by tnull** [comment]
> Can you give a sample `.bib` file that gives this error? Does it occur without the `--configfile`?

Yes, it also happens without the config file. I experimented a bit and found that the issue arises when two entries share the same non-present crossref. 

MWE:
```
@INPROCEEDINGS{one,
  AUTHOR = {Author One},
  CROSSREF = {crossref0},
  TITLE = {Paper One},
}

@INPROCEEDINGS{two,
  AUTHOR = {Author Two},
  CROSSREF = {crossref0},
  TITLE = {Paper Two},
}
```

**Comment by plk** [comment]
Should be fixed in 2.17 DEV.
